velox/exec/Spill.cpp

```diff
@@ -82,10 +100,11 @@ SpillState::SpillState(
       fileNamePrefix_(fileNamePrefix),
       maxPartitions_(maxPartitions),
       numSortKeys_(numSortKeys),
-      sortCompareFlags_(sortCompareFlags),
+      sortCompareFlags_(generateCompareFlags(sortCompareFlags, numSortKeys)),
```

Comment:
what's the problem with existing code so that we need this? Thanks!

Reply:
The sortCompareFlags may be empty when sort key exists. The RowContainer compareRows will check the sortCompareFlags size and use default constructor for it. https://github.com/facebookincubator/velox/blob/main/velox/exec/RowContainer.h#L726

But makePrefixSortLayout will check if the size if equal, so I make this change.